pkg/operator/platform/platform.go

```diff
@@ -59,3 +72,64 @@ func getClient(explicitKubeconfig string) (client.Client, error) {
 
 	return dynamicClient, nil
 }
+
+func GetFeatureGates(ctx context.Context) (featuregates.FeatureGate, error) {
+	stop := make(chan struct{})
+	ctx, cancelFn := context.WithCancel(ctx)
+	go func() {
+		defer cancelFn()
+		<-stop
+	}()
+
+	config, err := crtconfig.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kube config: %v", err)
+	}
+	clientSet, err := configclient.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	configInformers := configinformers.NewSharedInformerFactory(clientSet, 10*time.Minute)
+	desiredVersion := computeClusterOperatorVersions()
+	missingVersion := desiredVersion
```

Comment:
I believe this should be `missingVersion := "0.0.1-snapshot"` to match what is in the [03-deployment.yaml](https://github.com/openshift/cloud-credential-operator/blob/ee67cc6b1a8cbd2248a2536b0dfa93e18d1bfda2/manifests/03-deployment.yaml#L71)

Reply:
That seems wrong to hard code to a string which might end up different from YAML contents?

Reply:
Possibly... I just noted the difference from many of the other operators.

- [cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator/blob/ecb3786b1397cb93865952c98db72d1990587bd0/pkg/operator/operator.go#L106)
- [cloud-network-config-controller](https://github.com/openshift/cloud-network-config-controller/pull/102/files#diff-b52c6361b0946b90f2d35912dbe72c9dc8a5acc2d94a43cc26ad640b45e2be9aR41)
- [azure-disk-csi-driver-operator](https://github.com/openshift/azure-disk-csi-driver-operator/pull/83/files#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R108)
- [azure-file-csi-driver-operator](https://github.com/openshift/azure-file-csi-driver-operator/pull/54/files#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R74)
- [cluster-storage-operator](https://github.com/openshift/cluster-storage-operator/blob/ab03ebedd1b43342bf3ddd53cb113cf654f14d6d/pkg/operator/standalone_starter.go#L39)
- [machine-api-provider-azure](https://github.com/openshift/machine-api-provider-azure/blob/d3a1d2861718c6ae0ef0dbafaa7fe7bdbfa9fbef/cmd/manager/main.go#L137)

I believe what exists today (`missingVersion := desiredVersion`) will cause it to treat all versions as if they are the missing version, thereby skipping the version check altogether.

Reply:
Something to keep in mind, the purpose of `missingVersion` is to handle the situation where the 03-deployment.yaml is applied from locally compiled code, and where it is not going through the process of having a RELEASE_VERSION defined via our release process. Or, in other words, it is explicitly to enable locally compiled operator(s) to function as desired by ignoring the version check itself.

Reply:
👍